unstructured/file_utils/metadata.py

```diff
@@ -150,3 +153,43 @@ def _get_exif_datetime(exif_dict: Dict[str, Any], key: str) -> Optional[datetime
     # using the standard EXIF datetime format
     except ValueError:
         return None
+
+
+def apply_lang_metadata(
+    elements: Iterable[Element],
+    languages: List[str],
+    detect_language_per_element: bool = False,
+) -> Iterator[Element]:
+    """Detect and apply metadata.languages to each element in `elements`."""
+    # -- Note this function has a stream interface, but reads the full `elements` stream into memory
+    # -- before emitting the first updated element as output.
+
+    # Skip language detection for partitioners that use other partitioners.
+    # For example, partition_msg relies on partition_html and partition_text, but the metadata
+    # gets overwritten after elements have been returned by _html and _text,
+    # so `languages` would be detected twice.
+    if languages == [""]:
```

Comment:
So if partition_msg uses partition_html / _text, which `languages` result gets unused, and which one matches this "empty" case? Not sure I am follow how this check applies to the case where languages gets detected twice

Reply:
The languages detected by `partition_html/text` were unused as all of the metadata gets overwritten just after those partition calls (lines 105-111 in msg.py). You can see in there that I pass `languages=[""]` to those calls so we can just skip the language detection.